src/EnhancedSwitch.jsx

```diff
@@ -120,14 +120,23 @@ var EnhancedSwitch = React.createClass({
 
     var inputProps = {
       ref: "checkbox",
-      type: this.props.inputType,
       name: this.props.name,
-      value: this.props.value,
+      type: this.props.inputType,
       defaultChecked: this.props.defaultSwitched,
       onBlur: this._handleBlur,
       onFocus: this._handleFocus,
     };
 
+    /**
+     * If the input type is a checkbox, then we need to use the defaultValue instead of value
+     *  since in the reactjs v15.0.*, it only supports defaultValue
```

Comment:
Alright, can you research it a bit to better understand why? I don't think the comments are an accurate explanation of the problem and it could cause confusion later.